icefoganalytics/travel-authorization
# Travel Desk: Rank Travel Options – Fix Travel Port Interface

Relates to:

- https://github.com/icefoganalytics/travel-authorization/issues/230

# Context

Current "Travel Port" modal is unstable and hard to use as the end user, and as a developer.
This PR will rework the component to make it easier to use.

# Implementation

1. Rebuild "Travel Port" modal as TravelDeskFlightSegmentsManagePage.vue
2. Still the same frail importer, but now the code is more self contained.
3. Use session store to make imported, non-persisted data slight more resilient.
4. Rework flight segment group to be a more linear process, and persist data immediately.
5. Make it easier to delete individual segments.
6. Isolate flight option creation and edition into their own modals. Not nested modals either, just one layer.
7. Make it easier to delete individual flight groups.
8. Use newer standardized restful apis.
9. Implement flight segment re-order with `vuedraggable` and use dedicated and simple API endpoint for re-order. Reorders now persist immediately.
10. Clean up default layout and move primary nav to sidebar.

## Concerns

1. While I can now consistenly create/update/delete flight options and flight segments its very slow and I'd love to have more realistic training data. Also would love to rebuild the importer so it works with more than one example text.
2. There is a bunch of stuff that hasn't been implemented or tested thoroughly.
3. I'm not sure what effect the various travel desk status should have.

# Screenshots

Manage Groups/Flight Options
http://localhost:8080/travel-desk/1/manage-flight-segments
![image](https://github.com/user-attachments/assets/fd614ee6-b397-46fa-89cf-83d0aed8deca)

New Sidebar nav
![image](https://github.com/user-attachments/assets/25eb14fd-38c9-4ac5-8c50-4f06cc4c8acc)

Readonly travel desk request view
http://localhost:8080/travel-desk/1
![image](https://github.com/user-attachments/assets/f2661fe9-2659-401d-ae4c-68f33472b5ce)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
4. Boot the app via `dev up`
5. Log in to the app at http://localhost:8080
6. Create a travel authorization, and complete the process until you submit to the travel desk.
7. Log in as a travel desk user.
8. Go to the Travel Desk page from the sidebar nav.
9. Click on the "Edit" button to go to the travel request edit page.
10. Note that you can now save the various traveller details as part of the top section of the page.
    > Note: other than first section, all other content saves immediately.
11. Note that the question interface now works.
12. Note that the flight request UI now works.
13. You can now, create/update/delete flight requests via simple modals.
14. You can expand flight request to view flight options; there won't be any at first.
15. Scroll to the bottom of the flight request card, and click on the "manage flight options" button. I'm not sure if this should go hear or somewhere else?
16. Note that this button takes you to a new page.
17. Check that you can import the text from https://github.com/icefoganalytics/travel-authorization/pull/237. **NOTE** only this text works, until I get a chance to rebuild the parser.
18. Check that you can add/remove/edit flight segment information.
19. Check that you can group flight segments into a "flight option". Note that this persists the data.
20. Check that you can re-order flight segments in a flight option group.
21. Check that you can delete flight option groups.
22. Check that you edit flight option groups.
23. Complete everything you need to do, and the click the "return to travel request" button.
24. Scroll to the bottom of the page and click the "Send to Traveller" button.
25. Note that their is also a read-only version of the travel desk request accessible via the breadcrumbs.
